server/src/main/java/com/skillmagnet/Lesson/LessonController.java

```diff
@@ -0,0 +1,129 @@
+package com.skillmagnet.Lesson;
+
+import com.skillmagnet.Course.Course;
+import com.skillmagnet.Course.CourseRepository;
+import jakarta.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("api")
+public class LessonController {
+
+    @Autowired
+    LessonRepository lessonRepository;
+
+    @Autowired
+    CourseRepository courseRepository;
+
+    /**
+     * Retrieves a lesson by its ID.
+     *
+     * @param id The ID of the lesson to retrieve.
+     * @return {@code Lesson} on success otherwise a NOT_FOUND status.
+     */
+    @GetMapping("/lessons/{id}")
+    public ResponseEntity<?> getLesson(@PathVariable("id") int id) {
+        Optional<Lesson> lesson = lessonRepository.findById(id);
```

Comment:
Should I be using Optional<> ? What is the advantage? We can still check for null if it was just `Lesson lesson`? better standard?

Reply:
That's a good question and I don't really know. There are definitely cases and like code style guides that would suggest to always use Optionals. It can be super powerful / necessary in more complex spots. 

For how I'm using it here, it's acting no different than just doing `myObject != null` but the `lessonRepository` default `findById()` returns an optional thus why I'm using it lol